autoload/db/adapter/snowflake.vim

```diff
@@ -0,0 +1,45 @@
+let s:no_timing_friendly = ['-o', 'friendly=false', '-o', 'timing=false']
+
+function! s:command_for_url(url) abort
+  let url = db#url#parse(a:url)
+  "extra options here turn off nonsense before/after results
+  let cmd = (has_key(url, 'password') ? ['env', 'SNOWSQL_PWD=' . url.password] : []) +
+      \ [ "snowsql" ] +
```

Comment:
Hi, hope y'all don't mind if I chime in here a bit. I've been testing your branch to query Snowflake from `vim-dadbod` and `vim-dadbod-ui`, and here is some observations I made during usage:

## SSO

My company uses Okta to auth with Snowflake. This means I need to use the `--authenticator=externalbrowser` query param. When using `snowsql`, this means that a browser window pops up, and asks me to verify with Okta every time I want to perform a query. This is very frustrating.

I found that Snowflake is creating the `snow` CLI right now: https://docs.snowflake.com/en/developer-guide/snowflake-cli-v2/index

This tool has an automatic caching of the Okta credentials. So I only have to auth once, and every query I send afterwards does not force me to authorize again. Maybe using this newer tool would prove beneficial for SSO usecases? Let me know what you think. The tool is still experimental, but seems to work very well. Let me know if I can contribute.

## Email as username

This is a smal nit, but the `snowflake://username@accountname` querystring syntax doesn't work if your username is an email, just like in my case. I have to use `snowflake://accountname?username=<email>` instead. Not a big issue, just wanted to mention this.

Reply:
`snow` seems quite nice -- hadn't seen this yet. I can take a look at incorporating next week, or would accept a pr into my branch. zsh completion and credentials are two very nice QOL improvements.

Not sure how to solve the email as username issue given the syntax requirements, but it seems like that workaround is fine. Could maybe re-add support for the `snowflake://connection` syntax for folks who want to use global or workspace config file.

Reply:
> This is a smal nit, but the `snowflake://username@accountname` querystring syntax doesn't work if your username is an email, just like in my case.

https://github.com/tpope/vim-dadbod/blob/7888cb7164d69783d3dce4e0283decd26b82538b/doc/dadbod.txt#L47-L51

Reply:
Looking forward to using this with Snowflake support!